File: src/70. Climbing Stairs/70.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n < 3:
            return n
        list = [0] * (n+1)
        list[1] = 1
        list[2] = 2
        for i in range(3, n+1):
            list[i] = list[i-1] + list[i-2]
        return list[n]

    def climbStairs1(self, n: int) -> int:
        x = y = 1
        for i in range(n):
            x, y = x + y, x
        return y


if __name__ == "__main__":
    n = 5
    print(Solution().climbStairs(n))
    print(Solution().climbStairs1(n))

File: src/70. Climbing Stairs/70.java
class Solution {
    public int climbStairs(int n) {
        if (n == 1) {
            return 1;
        }
        int x = 1;
        int y = 2;
        for (int i = 3; i <= n; i++) {
            int temp = x + y;
            x = y;
            y = temp;
        }
        return y;
    }
    public static void main(String[] args) {
        int n = 3;
        System.out.println(n);
    }
}